### ISSUE TITLE
[REQUEST] dynamic batch size with gradient accumulate

### ISSUE THREAD

**Issue by Xiang-cd** [microsoft/DeepSpeed#6533]
**Is your feature request related to a problem? Please describe.**
my truly minimal use case request: I have 2 datasets with resolutions 256 and 512, I want to build 2 dataloaders, one dataloader load 
256x256 images with batch size 8, one load 512x512 images batch size 2.
it comes to a conflict with the note in the documentation:
```
Note: train_batch_size must be equal to train_micro_batch_size_per_gpu * gradient_accumulation_steps * number of GPUs. For simplicity, you can choose to only specify two of the three parameters, the last one will be inferred automatically by DeepSpeed.
```
so how to decide the train_micro_batch_size_per_gpu? 
that is comes to a grounding problem: how deepspeed process gradient accumulation?
1. judge the forward access of the model, no matter the batch size is what, the gradient accumulation is logical right
2. judge the instance number, namely the number of data instances go through the model, such as if instances reaches to 32 to perform optimization, so for 512x512 and 256X256 data, the forward times will be different, and cause a logical promblem with mixed batch size training.

**Describe the solution you'd like**
1. describe how deepspeed process gradient accumulation in the document
2. it is better to judge the forward access of the model to perform gradient accumulation
3. unlock the limitation of train_batch_size must be equal to train_micro_batch_size_per_gpu * gradient_accumulation_steps * number of GPUs

thank you for your great work.

**Comment by Xiang-cd** [comment]
hi, is there anyone?
